app/src/main/java/com/example/mp3demo/service/AudioPlayerService.java

```diff
@@ -0,0 +1,168 @@
+            mMediaPlayer = null;
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean start() {
+        if (null == mMediaPlayer) return false;
+        mMediaPlayer.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
+            @Override
+            public void onPrepared(MediaPlayer mp) {
+                mp.start();
+            }
+        });
+        mMediaPlayer.prepareAsync();
+        return true;
+    }
+
+    @Override
+    public boolean pause() {
+        if (null == mMediaPlayer || mMediaPlayer.isPlaying()) return false;
+        mMediaPlayer.pause();
+        sCurrentPosition = mMediaPlayer.getCurrentPosition();
+        return true;
+    }
+
+    @Override
+    public boolean stop() {
+        if (null == mMediaPlayer) return false;
+        mMediaPlayer.release();
+        return true;
+    }
+
+    @Override
+    public boolean play() {
+        if (null == mMediaPlayer) return false;
+        mMediaPlayer.start();
+        mMediaPlayer.seekTo(sCurrentPosition);
+        return true;
+    }
+
+    @Override
+    public Audio getCurrent() {
+        return sCurrentAudio;
+    }
+
+    @Override
+    public int getStatus() {
+        return (null == sCurrentAudio || null == mMediaPlayer) ? Constant.AUDIO_EMPTY
+                : mMediaPlayer.isPlaying() ? Constant.AUDIO_PLAYING
+                : Constant.AUDIO_PAUSE;
+    }
+
+    private void pushNotification(Audio audio) {
+        Intent notificationIntent = new Intent(this, AudiosActivity.class);
+        notificationIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        PendingIntent pendingIntent =
+                PendingIntent.getActivity(this, 0, notificationIntent, 0);
+
+        onCreateNotificationChannel();
+
+        Notification notification = getNotification(audio, pendingIntent);
+
+        startForeground(NOTIFICATION_ID, notification);
+    }
+
+    private Notification getNotification(Audio audio, PendingIntent pendingIntent) {
+        RemoteViews notificationLayout
+                = new RemoteViews(getPackageName(), R.layout.notification_audio_player);
+        notificationLayout.setTextViewText(R.id.text_title, audio.getTitle());
+        notificationLayout.setTextViewText(R.id.text_artist, audio.getArtist());
+        notificationLayout.setImageViewResource(R.id.image_control, R.drawable.ic_music_note_24dp);
+
+        return new NotificationCompat.Builder(this, CHANNEL_ID)
+                .setContentTitle(getText(R.string.notification_title))
+                .setContentText(getText(R.string.notification_message))
+                .setSmallIcon(R.drawable.ic_music_note_24dp)
+                .setCustomContentView(notificationLayout)
+                .setContentIntent(pendingIntent)
+                .setTicker(getText(R.string.ticker_text))
+                .setPriority(Notification.DEFAULT_ALL)
+                .build();
+    }
+
+    private void onCreateNotificationChannel() {
```

Comment:
```suggestion
    private void createNotificationChannel() {
```
Không phải callback thì không cần để on trước tên